sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorClient.cs

```diff
@@ -66,644 +56,316 @@ public MetricsAdvisorClient(Uri endpoint, TokenCredential credential)
         {
         }
 
-        /// <summary>
-        /// Initializes a new instance of the <see cref="MetricsAdvisorClient"/> class.
-        /// </summary>
-        /// <param name="endpoint">The endpoint to use for connecting to the Metrics Advisor Cognitive Service.</param>
-        /// <param name="credential">A credential used to authenticate to the service.</param>
-        /// <param name="options">A set of options to apply when configuring the client.</param>
-        /// <exception cref="ArgumentNullException"><paramref name="endpoint"/> or <paramref name="credential"/> is null.</exception>
-        public MetricsAdvisorClient(Uri endpoint, TokenCredential credential, MetricsAdvisorClientsOptions options)
```

Comment:
Why did we remove this constructor?

Reply:
It is removed because for DPG it is auto generated https://github.com/ShivangiReja/azure-sdk-for-net/blob/MetricsAdvisor-Experiment/sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Generated/MetricsAdvisorClient.cs#L43

Reply:
The generated method has a different method signature, though, so we have not preserved the API contract between HLC and Grown-up DPG.

Reply:
If we remove this constructor from the generated code then DPG will generate the default constructor:
```C#
public MetricsAdvisorClient(Uri endpoint, MetricsAdvisorClientsOptions options = null)
        {
            Argument.AssertNotNull(endpoint, nameof(endpoint));
            options ??= new MetricsAdvisorClientsOptions();

            ClientDiagnostics = new ClientDiagnostics(options);
            _pipeline = HttpPipelineBuilder.Build(options, Array.Empty<HttpPipelinePolicy>(), Array.Empty<HttpPipelinePolicy>(), new ResponseClassifier());
            _endpoint = endpoint;
        }
```

Are we fine with generating this public constructor?

Reply:
Can you help me understand why?  Wouldn't we still need credentials to connect to the service -- would this constructor be useful?  

I think this problem will go away when https://github.com/azure/autorest.csharp/issues/2105 is completed.

Reply:
Yes, we will need credentials to connect to the service and that's why we will add constructors with AAD and MetricsAdvisorKeyCredential in the convenience clients, just like HLC. 

But for DPG, if we don't add config like `security` we have added [here](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/deviceupdate/Azure.IoT.DeviceUpdate/src/autorest.md), then the generator will generate the default constructor , just like I mentioned above.

so I was just wondering, is it fine if we have this public constructor generated with `endpoint`? because we will add [these constructors](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorClient.cs#L33-L95) in the convenience client like HLC.

Reply:
> One ctor will take AADToken

Could this one be generated using `security: AADToken` in the config?

> One ctor will take endpoint (It will auto generated)

What is the scenario a customer would use this in?  We don't want to add constructors/methods that would never be used.

Reply:
>Could this one be generated using security: AADToken in the config?

Yes, currently I'm generating this. 

>What is the scenario a customer would use this in? We don't want to add constructors/methods that would never be used.

This is default constructor generated by DPG.  It is not generated currently in this experiment because I'm generating [ctor with AAD ](https://github.com/ShivangiReja/azure-sdk-for-net/blob/MetricsAdvisor-Experiment/sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Generated/MetricsAdvisorClient.cs#L43)

Reply:
https://github.com/azure/autorest.csharp/issues/2105 after this fix, constructor signature will be the same.